### PR TITLE
Do not allow modify aliases on followers

### PR DESCRIPTION
**Pull request by jasontedor**
Now that aliases are replicated by a follower from its leader, this commit prevents directly modifying aliases on follower indices.

Relates #41396

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/indices/alias/TransportIndicesAliasesAction.java [hunk 1]
```diff
@@ -37,15 +38,20 @@
 import org.elasticsearch.cluster.service.ClusterService;
 import org.elasticsearch.common.collect.ImmutableOpenMap;
 import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.index.Index;
 import org.elasticsearch.rest.action.admin.indices.AliasesNotFoundException;
 import org.elasticsearch.threadpool.ThreadPool;
 import org.elasticsearch.transport.TransportService;
 
 import java.util.ArrayList;
+import java.util.Arrays;
```
Comment (jasontedor) [inline server/src/main/java/org/elasticsearch/action/admin/indices/alias/TransportIndicesAliasesAction.java]:
```suggestion
```

File: server/src/main/java/org/elasticsearch/action/admin/indices/alias/TransportIndicesAliasesAction.java [hunk 2]
```diff
@@ -37,15 +38,20 @@
...
 import java.util.Collections;
 import java.util.HashSet;
 import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
 import java.util.Set;
+import java.util.stream.Collectors;
```
Comment (jasontedor) [inline server/src/main/java/org/elasticsearch/action/admin/indices/alias/TransportIndicesAliasesAction.java]:
```suggestion
```

**Review by dnhatn — APPROVED** [review]
LGTM. Should we consider this PR as a breaking-change for users won't be able to modify aliases on follower indices?

**Comment by jpountz** [comment]
@jasontedor Since this is a breaking change, could you add a mention to it in the migration guide for 7.3?

**Pull request merged**